src/rand.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Authors (alpha order): Adrian Catangiu @acatangiu, Alexandra Lordache @aghecenco, Alexandru Agache @alexandruag, Diana Popa @dianpopa, Matt Forrester @forbesmyester, Radu Weiss @raduweiss
+
+extern crate libc;
+use std::ffi::OsString;
+use std::str;
+
+pub fn timestamp_cycles() -> u64 {
+    #[cfg(target_arch = "x86_64")]
+    // Safe because there's nothing that can go wrong with this call.
+    unsafe {
+        std::arch::x86_64::_rdtsc() as u64
+    }
+    #[cfg(not(target_arch = "x86_64"))]
+    {
+        let mut ts = libc::timespec {
+            tv_sec: 0,
+            tv_nsec: 0,
+        };
+
+        unsafe {
+            libc::clock_gettime(libc::CLOCK_MONOTONIC, &mut ts);
+        }
+        (ts.tv_sec as u64) * 1000000000 + (ts.tv_nsec as u64)
+    }
+}
+
+fn timespec_to_us(time_struct: &libc::timespec) -> u64 {
+    (time_struct.tv_sec as u64) * 1_000_000 + (time_struct.tv_nsec as u64) / 1000
+}
+
+pub fn now_cputime_us() -> u64 {
+    let mut time_struct = libc::timespec {
+        tv_sec: 0,
+        tv_nsec: 0,
+    };
+    // Safe because the parameters are valid.
+    unsafe { libc::clock_gettime(libc::CLOCK_PROCESS_CPUTIME_ID, &mut time_struct) };
+    timespec_to_us(&time_struct)
+}
+
+
+// This generates pseudo random u32 numbers based on the current timestamp. Only works for x86_64,
```

Comment:
This comment can be changed to:
`\\\ Generate pseudo random u32 numbers based on the current timestamp.`
This will change the comment in a documentation comment.

Also, if this works only on x86_64 it should be labelled accordingly with `#[cfg(target_arch = "x86_64")]` and the comment about it only working on x86_64 should be deleted.

Reply:
Well this was/is it's state in `fc_util`, but I agree.

Reply:
I know this was as is in fc_util, but we have different requirements for things that end up in rust-vmm. One example is the mandatory documentation comment for public functions. If you need help with this let me know.

Reply:
I agreed and did it.